node/network/availability-distribution/src/lib.rs

```diff
@@ -417,77 +369,64 @@ where
 				// collect all direct interests of a peer w/o ancestors
 				state
 					.cached_live_candidates_unioned(view.0.iter())
-					.contains_key(&candidate_hash)
+					.contains(&candidate_hash)
 			})
 			.map(|(peer, _view)| peer.clone())
 			.collect();
 
+		let per_candidate = state.per_candidate.entry(candidate_hash).or_default();
+
+		let validator_count = per_candidate.validators.len();
+
 		// distribute all erasure messages to interested peers
 		for chunk_index in 0u32..(validator_count as u32) {
-			// only the peers which did not receive this particular erasure chunk
-			let per_candidate = state.per_candidate.entry(candidate_hash).or_default();
-
-			// obtain the chunks from the cache, if not fallback
-			// and query the availability store
-			let message_id = (candidate_hash, chunk_index);
-			let erasure_chunk = if let Some(message) = per_candidate.message_vault.get(&chunk_index) {
-				message.erasure_chunk.clone()
+			let message = if let Some(message) = per_candidate.message_vault.get(&chunk_index) {
+				tracing::trace!(
+					target: LOG_TARGET,
+					%chunk_index,
+					?candidate_hash,
+					"Retrieved chunk from message vault",
+				);
+				message.clone()
 			} else if let Some(erasure_chunk) = query_chunk(ctx, candidate_hash, chunk_index as ValidatorIndex).await? {
-				erasure_chunk
+				tracing::trace!(
+					target: LOG_TARGET,
+					%chunk_index,
+					?candidate_hash,
+					"Retrieved chunk from availability storage",
+				);
+
+				AvailabilityGossipMessage {
+					candidate_hash,
+					erasure_chunk,
+				}
 			} else {
-				continue;
+				tracing::error!(
+					target: LOG_TARGET,
+					%chunk_index,
+					?candidate_hash,
+					"Availability store reported that we have the availability data, but we could not retrieve a chunk of it!",
+				);
+				break;
```

Comment:
This became `break` that was a `continue`. Should we not try to retrieve as many chunks as possible?

Reply:
Reverted.